data_collection/gazette/spiders/rj_sao_joao_de_meriti.py

```diff
@@ -0,0 +1,109 @@
+import datetime
+import json
+import re
+
+import scrapy
+from dateutil.rrule import MONTHLY, rrule
+
+from gazette.items import Gazette
+from gazette.spiders.base import BaseGazetteSpider
+
+
+class SpSaoJoaoDeMeritiSpider(BaseGazetteSpider):
+    TERRITORY_ID = "3305109"
+    name = "rj_sao_joao_de_meriti"
+    allowed_domains = ["meriti.rj.gov.br"]
+
+    URL_BEFORE_2018 = "https://meriti.rj.gov.br/dom_mp/"
+    URL_FROM_2018 = (
+        "https://transparencia.meriti.rj.gov.br/diario_oficial_get.php?mesano="
+    )
+
+    start_date = datetime.date(2011, 9, 30)
+    end_date = datetime.date.today()
+
+    def start_requests(self):
+        janeiro_2018 = datetime.date(2018, 1, 1)
+        for month in self.months_interval():
+            if month < janeiro_2018:
+                yield scrapy.Request(
+                    f"{self.URL_BEFORE_2018}{month.year}{month.month:02}.php",
+                    callback = self.parse_month_before_2018
+                )
+            else:
+                yield scrapy.Request(
+                    f"{self.URL_FROM_2018}{month.month}/{month.year}",
+                    callback = self.parse_month_from_2018
+                )
+
+    def months_interval(self):
+        month_rule = rrule(freq=MONTHLY, dtstart=self.start_date, until=self.end_date)
+        months = [month.date() for month in month_rule]
+        if months and months[-1].month != self.end_date.month:
+            months.append(self.end_date)
+        return months
+
+    def get_day_fom_pdf_link(self, pdf_link):
+        day_regex = r"[/-](\d{1,2})[^\d/][^/]*.pdf"
+        day = int(re.search(day_regex, pdf_link, re.IGNORECASE).group(1))
+        return day
+
+    def parse_gazette_before_2018(self, month_response, pdf_link):
+        year = int(month_response.url[-10:-6])

```

Comment:
**suggestion**: You already have the year and the month when you create this request, so you could pass it as a `meta` argument or a `cb_kwargs`. This is more reliable as you don't rely on magic numbers to obtain this value, so if they change the URL (even one char is enough) this will break.